core/settings/qPrefUnit.h

```diff
@@ -14,7 +14,7 @@ class qPrefUnits : public QObject {
 	Q_PROPERTY(QString pressure READ pressure WRITE set_pressure NOTIFY pressureStringChanged)
 	Q_PROPERTY(bool show_units_table READ show_units_table WRITE set_show_units_table NOTIFY show_units_tableChanged)
 	Q_PROPERTY(QString temperature READ temperature WRITE set_temperature NOTIFY temperatureStringChanged)
-	Q_PROPERTY(QString unit_system READ unit_system WRITE set_unit_system NOTIFY unit_systemChanged)
+	Q_PROPERTY(QString unit_system READ unit_system WRITE set_unit_system NOTIFY unit_systemStringChanged)
```

Comment:
doesn't that remove the `enum` based version?
or is this something where both flavors work because of overloading? In that case, why does the signal name have to change?

Reply:
No we never used the enum here. I am changing the signal to <name>StringChanged(value) to be consistent with the other signals.

Normally the signal name would be the same and simply have different parameters. But having overloaded functions which causes problems in connect, and forced me to use the SIGNAL/SLOT macros, was something you did not accept, hence 2 different function names.

If you have changed your mind, I will make a patch PR afterwards and correct all of qPrefUnit in one go.

Reply:
I think I followed up when we went through this and said something to the extend of "oh, with that explanation this makes sense" and then we talked about how you sometimes don't advocate well for your changes :-)
I am OK with the `SIGNAL()/SLOT()` syntax where it provides clear value. I.e. if switching to the new syntax causes ugly problems, then don't. This here is an example where using the new syntax appears to make things more complicated - in another commit you worked around that with a lambda function. If that's not an option here, then feel free to go with the old syntax.